failmsg.go

```diff
@@ -82,3 +84,11 @@ func (msg FailureMessage) Prefix(s string) FailureMessage {
 	}
 	return FailureMessage(s) + msg
 }
+
+// AsError returns the failure message as an error type, or nil if the message is empty.
+func (msg FailureMessage) AsError() error {
+	if msg == "" {
+		return nil
+	}
+	return fmt.Errorf(string(msg))
```

Comment:
What do you think of introducing an `AssertionError` type like following

```go
type AssertionError struct {
  Message FailureMessage
}

func (err *AssertionError) Error() string {
  return msg
}
```

and then using it like this

```suggestion
	return &AssertionError{Message: msg}
```